### ISSUE TITLE
Encoded password does not look like BCrypt

### ISSUE THREAD

**Issue by arashkhalesi** [szerhusenBC/jwt-spring-security-demo#37]
I generated hash password in https://bcrypt-generator.com/
Client: Postman 

DB password:        :$ 2y$10$cZWIQLCT.AUu6/f.54ri/uX5VeMAANDY1f3wLh8NpVi6.mljPh7DO

enc                        : $2y$10$cZWIQLCT.AUu6/f.54ri/uX5VeMAANDY1f3wLh8NpVi6.mljPh7DO
raw                        : $2y$10$cZWIQLCT.AUu6/f.54ri/uX5VeMAANDY1f3wLh8NpVi6.mljPh7DO
Password param   : $2y$10$cZWIQLCT.AUu6/f.54ri/uX5VeMAANDY1f3wLh8NpVi6.mljPh7DO
----------------------------------------------------------------------------------

Encoded password does not look like BCrypt
I debugged it and it failed in if (!BCRYPT_PATTERN.matcher(encodedPassword).matches()) {
Both password is the same I double checked them in debug mode

spring.BCryptPasswordEncoder.java
public boolean matches(CharSequence rawPassword, String encodedPassword) {
		if (encodedPassword == null || encodedPassword.length() == 0) {
			logger.warn("Empty encoded password");
			return false;
		}
		
		
		// fialed here : encodedPassword is not match with BCRYPT_PATTERN

		if (!BCRYPT_PATTERN.matcher(encodedPassword).matches()) {
			logger.warn("Encoded password does not look like BCrypt");
			return false;
		}

		return BCrypt.checkpw(rawPassword.toString(), encodedPassword);
	}

I just hash a raw password with EncodedPassword from Spring but the same issue was faced. 
Any suggestion ?

**Comment by arashkhalesi** [comment]
Fixed. I removed @EnableGlobalMethodSecurity and instantiated explicitly in WebConfig.

**Comment by miloradtrninic** [comment]
I have the same issue. But I don't understand your explanation. Can you please elaborate?
